pkg/cmd/roachprod/install/cockroach.go

```diff
@@ -396,19 +396,86 @@ func (h *crdbInstallHelper) generateStartCmd(
 	binary := cockroachNodeBinary(h.c, nodes[nodeIdx])
 	keyCmd := h.generateKeyCmd(nodeIdx, extraArgs)
 
-	// NB: this is awkward as when the process fails, the test runner will show an
-	// unhelpful empty error (since everything has been redirected away). This is
-	// unfortunately equally awkward to address.
+	if h.c.IsLocal() {
```

Comment:
We shouldn't let these diverge as much. Of course we can't ask for systemd to be installed locally, but can we share the `run-cockroach.sh` script and make the only difference that in `local` we invoke it with a `--background` arg directly?